src/test-run/index.js

```diff
@@ -84,7 +84,8 @@ export default class TestRun extends Session {
         this.fileDownloadingHandled               = false;
         this.resolveWaitForFileDownloadingPromise = null;
 
-        this.debugging = false;
+        this.debugging              = false;
+        this.previousDebuggingState = false;
```

Comment:
It would be nice to avoid class-level variable here. Let use local var and just pass it to `getSnapshotFromRole`